dbt/adapters/base/impl.py

```diff
@@ -1574,13 +1579,24 @@ def valid_incremental_strategies(self):
 
     def builtin_incremental_strategies(self):
         builtin_strategies = ["append", "delete+insert", "merge", "insert_overwrite"]
-        if os.environ.get("DBT_EXPERIMENTAL_MICROBATCH"):
+        if self.behavior.require_batched_execution_for_custom_microbatch_strategy:
```

Comment:
Adding `.no_warn` here would make it so that the warning doesn't fire on every incremental model. But then we would need to evaluate this deeper in the code.